### DISCUSSION TITLE
Nordpool api token ?

### DISCUSSION THREAD

**Discussion by RTTTC** [custom-components/nordpool#312] [General]
We are trying to integrate Nordpool data into evcc EV charge controller, and it seems normally Nordpool needs access credentials, so how does your integration work around it? https://github.com/evcc-io/evcc/discussions/1545#discussioncomment-5614882

**Comment by Hellowlol** [comment]
Hi,

Technically there is no need for an API key to get the data. The code used to get the data is https://github.com/custom-components/nordpool/blob/master/custom_components/nordpool/aio_price.py

Im currently working on creating a free API that does not require API keys, without polling data from nordpool. Instead, it will get the data from entsoe using my API key and cache the results locally after some conventions are done.

**Reply by RTTTC** [reply]
ENTSO-E is even better!! 😎 More countries. Do you gave a timeline in mind?

**Reply by RTTTC** [reply]
@Hellowlol How is the ENSTO-E integration going? Any updates?

**Comment by Hellowlol** [comment]
mpv is working. Just need to decide on how it should be hosted

**Comment by large** [comment]
So will this still work in 2024 when nordpool changes their portal? I guess the api has been open since their old portal needed it. (Best guess...)

New portal is https://data.nordpoolgroup.com and they are using angular, whole different ball-game.
